### PR TITLE
change: [M3-8134] - Take two of `vitest-preview`

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
> [!NOTE]
> This is a re-attempt of #10481, which was reverted since it included breaking config changes.
> This attempt does not modify the production `vite.config.ts` and isolates the necessary changes to the `vitest-preview` server itself.

## Description 📝
[`vitest-preview`](https://www.vitest-preview.com/) is a nifty package that lets you visually debug unit tests in the browser. I believe it could be a useful addition for our unit testing workflow.

## Changes  🔄
- Add `vitest-preview` package and make related configuration changes
- Document how to use it to visually debug unit tests

## How to test 🧪
- Follow the steps in the documentation to debug a test using vitest-preview
- Review documentation changes for readability, concision, and grammar/spelling errors

**Verify the unintended side-effects from the previous iteration of this PR have been resolved:**
- Running the dev server works and does not result in changes to `yarn.lock`

### CODE REVIEW

**Comment by mjac0bs** [comment]
I think it's worth creating a ticket for yourself for this @hkhalil-akamai, especially since had some follow up investigation that you put in.

File: scripts/vitest-preview/start-preview-server.mjs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      vite.ssrFixStacktrace(e);
+      next(e);
+    }
+  });
+  app.listen(port, () => {
+    console.log(`Vitest Preview Server listening on http://localhost:${port}`);
+    openBrowser(`http://localhost:${port}`);
+  });
+}
+createServer();
```
Comment (abailly-akamai) [inline scripts/vitest-preview/start-preview-server.mjs:57]: This is probably fine, but I somehow have qualms committing this to our repo. Maybe i am just paranoid at this point 🤣 - have you tried TS for it?

File: yarn.lock [hunk 2]
```diff
@@ -13488,6 +13637,18 @@ vite-plugin-svgr@^3.2.0:
...
+  resolved "https://registry.yarnpkg.com/vite/-/vite-3.2.10.tgz#7ac79fead82cfb6b5bf65613cd82fba6dcc81340"
+  integrity sha512-Dx3olBo/ODNiMVk/cA5Yft9Ws+snLOXrhLtrI3F4XLt4syz2Yg8fayZMWScPKoz12v5BUv7VEmQHnsfpY80fYw==
+  dependencies:
+    esbuild "^0.15.9"
+    postcss "^8.4.18"
+    resolve "^1.22.1"
+    rollup "^2.79.1"
+  optionalDependencies:
+    fsevents "~2.3.2"
+
```
Comment (abailly-akamai) [inline yarn.lock:13651]: Well that's a pity

**Review by abailly-akamai — COMMENTED** [review]
Thanks for this new version!

Looking at this I wonder about building our own, even if based on this (aka we fork our own and updates a bunch of dependencies), rather than keeping the package and overriding bits of it.

This is version `0.0.1` last updated in 2022, which means we're unlikely to see updates and we will be stuck with old deps and resolutions in our lock. Not the end of the world but the lack of support for this makes it less appealing looking closely at it.

I won't stop the train if we wanna go that route cause this tool has good benefits but wanted to plant a warning seed if doing so.

**Review by dwiley-akamai — COMMENTED** [review]
Was able to get the preview up by following the documentation ✅

As Alban mentioned, it's worth considering forking the project. It appears the creator hasn't been active in the repo since last summer, commenting on an open issue.

**Comment by hkhalil-akamai** [comment]
Closed as I investigate forking or re-implementing the package.

**Pull request closed without merging**